cookbook/upgrade/major_version.rst

```diff
@@ -35,13 +35,27 @@ functionality are removed. So, as long as you've updated your code to stop
 using these deprecated features in the last version before the major (e.g.
 2.8.*), you should be able to upgrade without a problem.
 
-To help you with this, the last minor releases will trigger deprecated notices.
-For example, 2.7 and 2.8 trigger deprecated notices. When visiting your
-application in the :doc:`dev environment </cookbook/configuration/environments>`
+To help you with this, deprecation notices are triggered whenever you end up
+using a deprecated feature. When visiting your application in the
```

Comment:
do we know immediately trigger deprecated notices whenever something is deprecated? (in 3.1 for example?) Otherwise, I propose to readd the versions in this sentence.